### PR TITLE
allow customize export encoding

### PR DESCRIPTION
**Pull request by exherb**

### CODE REVIEW

**Comment by exherb** [comment]
please review~

File: src/client/utils/download/download.ts [hunk 1]
```diff
@@ -38,12 +38,24 @@ export function getMIMEType(fileType: string) {
...
+  function save(part: string | Buffer) {
+    const type = `${getMIMEType(fileFormat)};charset=${encoding}`;
+    const blob = new Blob([part], { type });
+    if (!fileName) fileName = `${new Date()}-data`;
+    fileName += `.${fileFormat}`;
+    filesaver.saveAs(blob, fileName, true); // true == disable auto BOM
+  }
+
+  const result = datasetToFileString(dataset, fileFormat, options);
+  encoding = encoding === "" ? "utf-8" : encoding;
```
Comment (adrianmroz-allegro) [inline src/client/utils/download/download.ts:51]: There's already a default `utf-8` in function declaration, this line can be removed. `""` is just an invalid value like `"foobar"`, we shouldn't treat it differently.

File: src/client/utils/download/download.ts [hunk 2]
```diff
@@ -38,12 +38,24 @@ export function getMIMEType(fileType: string) {
...
+  if (encoding !== "utf-8") {
+    import("iconv-lite").then(function(iconv) {
```
Comment (adrianmroz-allegro) [inline src/client/utils/download/download.ts:53]: Nice idea with dynamic import. Does current build really puts `iconv-lite` in different bundle?

**Review by adrianmroz-allegro — COMMENTED** [review]
We need to decide where this setting belongs. We are thinking about adding `locale` section in `customisation` and this is good candidate.

File: src/client/utils/download/download.ts (hunk 1 above)
Comment (exherb) [reply src/client/utils/download/download.ts:51]: cool

File: src/client/utils/download/download.ts (hunk 2 above)
Comment (exherb) [reply src/client/utils/download/download.ts:53]: yes

**Comment by exherb** [comment]
> locale

like this?

```yaml
customization:
  locale:
    encoding: gbk
```

File: src/client/utils/download/download.ts (hunk 1 above)
Comment (exherb) [reply src/client/utils/download/download.ts]: resolved, thanks

**Comment by adrianmroz-allegro** [comment]
Hi! I added correct fields in #743. Would you mind if I use your code and glue it together in new pull request?

Sorry for late response, but OAuth support took longer that we anticipated.

**Comment by exherb** [comment]
> Hi! I added correct fields in #743. Would you mind if I use your code and glue it together in new pull request?
> 
> Sorry for late response, but OAuth support took longer that we anticipated.

ok. thanks

**Comment by exherb** [comment]
> Hi! I added correct fields in #743. Would you mind if I use your code and glue it together in new pull request?
> 
> Sorry for late response, but OAuth support took longer that we anticipated.

we use oauth gateway …

**Comment by exherb** [comment]
> Hi! I added correct fields in #743. Would you mind if I use your code and glue it together in new pull request?
> 
> Sorry for late response, but OAuth support took longer that we anticipated.

seems lost contributor info.

**Comment by adrianmroz-allegro** [comment]
Yeah, I added commit and obviously it is my name in git log. 

Nevertheless it is my mistake and I should acknowledge your contribution. How can I make it right with you?

**Comment by exherb** [comment]
It’s alright

**Pull request closed without merging**